### PR TITLE
Chore: Deprecate components available in @grafana/ui

### PR DESCRIPTION
**Pull request by Clarity-89**
- Add deprecation warnings to the components that are available in grafana/ui to reduce the imports of those components from this package. 
- Remove the `SecretInput` and `SecretTextArea` components.

### CODE REVIEW

File: src/ConfigEditor/Auth/common/SecretInput.tsx [hunk 1]
```diff
@@ -17,6 +17,9 @@ export type Props = React.ComponentProps<typeof Input> & {
 export const CONFIGURED_TEXT = 'configured';
 export const RESET_BUTTON_TEXT = 'Reset';
 
+/**
+ * @deprecated use the component from @grafana/ui instead. Available starting from @grafana/ui@9.2
+ */
```
Comment (Clarity-89) [inline src/ConfigEditor/Auth/common/SecretInput.tsx:22]: @cletter7 do we still need to support Grafana 8 anywhere or can these components be removed from here?

File: src/QueryEditor/Space.tsx [hunk 2]
```diff
@@ -10,6 +10,9 @@ export interface SpaceProps {
   layout?: 'block' | 'inline';
 }
 
+/**
+ * @deprecated use the Space component from @grafana/ui instead. Available starting from @grafana/ui@10.4.0
```
Comment (Clarity-89) [inline src/QueryEditor/Space.tsx:14]: 10.4 is not released yet, but I've added this here to show that an alternative is available.

File: src/ConfigEditor/Auth/common/SecretInput.tsx (hunk 1 above)
Comment (cletter7) [reply src/ConfigEditor/Auth/common/SecretInput.tsx:22]: I think we can remove them. In enterprise datasources we now support grafana 9.5+

**Review by ivanahuckova — APPROVED** [review]
LGTM

**Pull request merged**